### ISSUE TITLE
Logging configuration not recognizing the classes' package

### ISSUE THREAD

**Issue by x80486** [apache/camel-quarkus#6881]
## Description

I'm facing a small problem with the logging output. The current logging configuration is basically this:

```yaml
quarkus:
  ...
  log:
    category:
      "io.quarkus":
        level: INFO
      "tld.domain
        level: INFO
    console:
      async: true
      enable: true
      format: "%d{yyyy-MM-dd HH:mm:ss,SSS} |- %-5p in %c:%L [%t] - %m%e%n"
      json: false
    level: WARN
```

With this, I'm expecting that any classes under `tld.domain` will be logged at `INFO` level, but the application does not log any output for these classes unless I set `log.level` to a higher threshold.

Moreover, based on the format that I have, the class names and their packages must be logged, but with Camel only the class name is logged — and I guess that may be the actual issue.

This is the application output when I set `log.level` to `INFO`:

```shell
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2024-12-26 23:26:48,519 |- INFO  in org.apache.camel.quarkus.core.CamelBootstrapRecorder:46 [Quarkus Main Thread] - Apache Camel Quarkus 3.17.0 is starting
2024-12-26 23:26:48,520 |- INFO  in org.apache.camel.main.MainSupport:71 [Quarkus Main Thread] - Apache Camel (Main) 4.9.0 is starting
2024-12-26 23:26:48,527 |- INFO  in org.apache.camel.main.ProfileConfigurer:97 [Quarkus Main Thread] - The application is starting with profile: dev
2024-12-26 23:26:48,622 |- INFO  in org.apache.camel.impl.engine.AbstractCamelContext:2524 [Quarkus Main Thread] - Apache Camel 4.9.0 (camel-1) is starting
2024-12-26 23:26:48,626 |- INFO  in org.apache.camel.impl.engine.AbstractCamelContext:2727 [Quarkus Main Thread] - Routes startup (total:1)
2024-12-26 23:26:48,627 |- INFO  in org.apache.camel.impl.engine.AbstractCamelContext:2731 [Quarkus Main Thread] -     Started timer.route (timer://sampleTimer)
2024-12-26 23:26:48,627 |- INFO  in org.apache.camel.impl.engine.AbstractCamelContext:2764 [Quarkus Main Thread] - Apache Camel 4.9.0 (camel-1) started in 4ms (build:0ms init:0ms start:4ms boot:380ms)
2024-12-26 23:26:48,703 |- INFO  in io.quarkus:109 [Quarkus Main Thread] - application 0.0.0-SNAPSHOT on JVM (powered by Quarkus 3.17.4) started in 1.901s. Listening on: http://localhost:8080
2024-12-26 23:26:48,704 |- INFO  in io.quarkus:113 [Quarkus Main Thread] - Profile dev activated. Live Coding activated.
2024-12-26 23:26:48,704 |- INFO  in io.quarkus:115 [Quarkus Main Thread] - Installed features: [camel-core, camel-log, camel-timer, cdi, config-yaml, smallrye-context-propagation, vertx]
2024-12-26 23:26:49,633 |- INFO  in ExampleRouteBuilder:14:166 [Camel (camel-1) thread #1 - timer://sampleTimer] - Checking the gravitational constant in your locale...
2024-12-26 23:26:52,626 |- INFO  in ExampleRouteBuilder:14:166 [Camel (camel-1) thread #1 - timer://sampleTimer] - Checking the gravitational constant in your locale...
q
2024-12-26 23:26:53,681 |- INFO  in org.apache.camel.impl.engine.AbstractCamelContext:2981 [Shutdown thread] - Apache Camel 4.9.0 (camel-1) is shutting down
2024-12-26 23:26:53,689 |- INFO  in org.apache.camel.impl.engine.AbstractCamelContext:166 [Shutdown thread] - Routes stopped (total:1)
2024-12-26 23:26:53,689 |- INFO  in org.apache.camel.impl.engine.AbstractCamelContext:166 [Shutdown thread] -     Stopped timer.route (timer://sampleTimer)
2024-12-26 23:26:53,693 |- INFO  in org.apache.camel.impl.engine.AbstractCamelContext:3112 [Shutdown thread] - Apache Camel 4.9.0 (camel-1) shutdown in 11ms (uptime:5s)
2024-12-26 23:26:53,701 |- INFO  in io.quarkus:124 [Shutdown thread] - application stopped in 0.022s
```

And what I was mentioning is that these log statements should have the package names: `tld.domain.route.ExampleRouteBuilder` instead:

```shell
2024-12-26 23:26:49,633 |- INFO  in ExampleRouteBuilder:14:166 [Camel (camel-1) thread #1 - timer://sampleTimer] - Checking the gravitational constant in your locale...
2024-12-26 23:26:52,626 |- INFO  in ExampleRouteBuilder:14:166 [Camel (camel-1) thread #1 - timer://sampleTimer] - Checking the gravitational constant in your locale...
```

By the way, the same logging configuration I use it in almost all other Quarkus applications, that's why I think this may be something with Camel.

**Comment by zhfeng** [comment]
It looks like you are using `camel-log` and it has a different format, see https://camel.apache.org/components/4.8.x/log-component.html#_full_customization_of_the_logging_output

**Comment by x80486** [comment]
Yes, I'm using `camel-log`. Thanks for the clarification. I'll see what I can do to align these with the overall application configuration.

Initially I was under the impression that `camel-log` would pick up the global logging configuration when using with `Quarkus` or `Spring Boot`.

**Comment by kubiani** [comment]
Was having the same issue. I think I finally figured it out. 

TL;DR: I was also assuming the category was/also accepted the package; this doesn't appear to be the case. Camel default appears to set the category as `<ClassName>`:`<LineNum>`. This is also the `%c` qualifier in the format so `ExampleRouteBuilder:14` in your example. 

So per your example, (I think) the correct logging configuration to get the desired result would be:
```yaml
quarkus:
  log:
    category:
      "ExampleRouteBuilder*":  # the * suffix is important
        level: INFO
    level: WARN
```

This assumes you have a class along the lines of: 
```java
package tld.domain;

public class ExampleRouteBuilder extends RouteBuilder {

  @Override
  public void configure() throws Exception {

    // any route defined in here will be categorized as ExampleRouteBuilder with a corresponding line id
    from("direct:a").to("direct:b");
    from("direct:c").to("direct:d");

  }
}
```

This intuitively makes sense, but none of the documentation I've come across makes this clear.

**Comment by x80486** [comment]
For what it's worth, using `.log(/* org.apache.camel */ LoggingLevel.DEBUG, logger, "Hello, Camel")` will work as expected.

Here, `logger` refers to the actual SLF4J instance used by the application (not the default `log`), and the log level must be equal to or higher than the level defined for the corresponding category.
